### PR TITLE
CLOUDP-72402: Add e2e tests for atlas integrations

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-72402

Closes #[issue number]

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: e2e/atlas/integrations_test.go [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+			victorOpsEntity,
+			"--apiKey",
+			key,
+			"-o=json")
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+
+		a := assert.New(t)
+		a.NoError(err, string(resp))
+	})
```
Comment (andreaangiolillo) [inline e2e/atlas/integrations_test.go:222]: VICTOR_OPS is not inside the returned list. I have created this task to investigate this behaviour [CLOUDP-71439](https://jira.mongodb.org/browse/CLOUDP-71439)

File: e2e/atlas/integrations_test.go [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+	datadogEntity      = "DATADOG"
+	flowdockEntity     = "FLOWDOCK"
+	newRelicEntity     = "NEW_RELIC"
+	opsGenieEntity     = "OPS_GENIE"
+	pagerDutyEntity    = "PAGER_DUTY"
+	victorOpsEntity    = "VICTOR_OPS"
+	webhookEntity      = "WEBHOOK"
+)
+
+func TestIntegrations(t *testing.T) {
```
Comment (gssbzn) [inline e2e/atlas/integrations_test.go:41]: one major thing here, we may get false positives from running this test in parallel given there can only be one of these integrations per project and we're always running with the same project
I think we should reuse your helper to create a new project here and limit the scope of the test and avoid false possitives

File: e2e/atlas/integrations_test.go [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+		var thirdPartyIntegrations mongodbatlas.ThirdPartyIntegrations
+		if err := json.Unmarshal(resp, &thirdPartyIntegrations); a.NoError(err) {
+			found := false
+			services := thirdPartyIntegrations.Results
+			for i := range services {
+				if services[i].Type == datadogEntity {
+					found = true
+					break
+				}
+			}
```
Comment (gssbzn) [inline e2e/atlas/integrations_test.go:77]: I think we should create a helper `integrationExists(name string) bool` and reuse it for these create tests

**Review by gssbzn — CHANGES_REQUESTED** [review]
We may face some false positives here given we can only have one integration type per project so we should try to scope this test to run on its own new project

**Comment by andreaangiolillo** [comment]
I am not sure what this error means
```
e2e/atlas/helper_test.go:145:97: integrationExists - result 0 (bool) is never used (unparam)
func integrationExists(name string, thirdPartyIntegrations mongodbatlas.ThirdPartyIntegrations) bool {
```

File: e2e/atlas/integrations_test.go [hunk 4]
```diff
@@ -100,15 +109,7 @@ func TestIntegrations(t *testing.T) {
...
-			found := false
-			services := thirdPartyIntegrations.Results
-			for i := range services {
-				if services[i].Type == flowdockEntity {
-					found = true
-					break
-				}
-			}
-			a.True(found)
+			integrationExists(flowdockEntity, thirdPartyIntegrations)
```
Comment (gssbzn) [inline e2e/atlas/integrations_test.go:112]: the error is you never use the return value of the function, here you're missing a `a.True(integrationExists(...))`

File: e2e/atlas/integrations_test.go [hunk 5]
```diff
@@ -50,6 +50,19 @@ func TestIntegrations(t *testing.T) {
 	if err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
+
+	projectName := fmt.Sprintf("e2e-proj-%v", n)
```
Comment (gssbzn) [inline e2e/atlas/integrations_test.go:54]: [nit] can we rename this, sometimes test left things around in atlas and this way we can better diagnose where are we leaking data
```suggestion
	projectName := fmt.Sprintf("e2e-integration-proj-%v", n)
```

**Review by gssbzn — CHANGES_REQUESTED** [review]
One small nit and let's fix that linting error

**Review by gssbzn — APPROVED** [review]
LGTM thanks for adding these tests

**Pull request merged**